phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -426,58 +421,39 @@ RegionScanner getWrappedScanner(final ObserverContext<RegionCoprocessorEnvironme
     public void preCompactScannerOpen(ObserverContext<RegionCoprocessorEnvironment> c, Store store,
                                              ScanType scanType, ScanOptions options, CompactionLifeCycleTracker tracker,
                                              CompactionRequest request) throws IOException {
-        Configuration conf = c.getEnvironment().getConfiguration();
-        if (isMaxLookbackTimeEnabled(conf)) {
-            setScanOptionsForFlushesAndCompactions(conf, options, store, scanType);
-        }
+        setScanOptionsForFlushesAndCompactions(options);
     }
 
     @Override
     public void preFlushScannerOpen(ObserverContext<RegionCoprocessorEnvironment> c, Store store,
                                                 ScanOptions options, FlushLifeCycleTracker tracker) throws IOException {
-        Configuration conf = c.getEnvironment().getConfiguration();
-        if (isMaxLookbackTimeEnabled(conf)) {
-            setScanOptionsForFlushesAndCompactions(conf, options, store, ScanType.COMPACT_RETAIN_DELETES);
-        }
+        setScanOptionsForFlushesAndCompactions(options);
```

Comment:
Should the preFlush also have a specialized scanner like the StoreCompactionScanner (or maybe can be reused)?.
As I understand the preFlush currently will keep all rows, versions and delete markers.

Reply:
Keeping all versions and delete markers is the required behavior for memstore flushes. That is required for indexing consistency and also for max lookback.

Reply:
But doesn't running the flush thru the StoreCompactionScanner produce the same outcome, i.e keeping the right row versions taking maxLookback and delete markers and all the other settings into account?

Reply:
But then why to run StoreCompactionScanner if it does not make any difference in this case? The timestamp of these rows in memory will be very recent and they will not expire in practice.